content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+      </ul></td>
+    </tr>
+    <tr>
+      <td><a href="/docs/reference/access-authn-authz/extensible-admission-controllers/">Validating admission webhook</a></td>
+      <td>Intercept API requests before admission and validate against complex
+        policy declarations.</td>
+      <td><ul>
+        <li>Validate critical configurations before resource admission.</li>
+        <li>Enforce complex policy logic before admission.</li>
+      </ul></td>
+    </tr>
+    <tr>
+      <td><a href="/docs/reference/access-authn-authz/validating-admission-policy/">Validating admission policy</a></td>
+      <td>Intercept API requests before admission and validate against CEL
+        expressions.</td>
+      <td><ul>
+        <li>Validate critical configurations before resource admission.</li>
+        <li>Enforce policy logic using CEL expressions.</li>
+      </ul></td>
+    </tr>
+  </tbody>
+</table>
+
+In general, use _webhook_ admission control when you want an extensible way to
+declare or configure the logic. Use built-in CEL-based admission control when
+you want to declare simpler logic without the overhead of running a webhook
+server. The Kubernetes project recommends that you use CEL-based admission
+control when possible.
+
+### Use built-in validation and defaulting for CustomResourceDefinitions {#no-crd-validation-defaulting}
+
+If you use
+{{< glossary_tooltip text="CustomResourceDefinitions" term_id="customresourcedefinition" >}},
+don't use mutating webhooks to validate values in CustomResource specifications
+or to set default values for fields. Kubernetes lets you define validation rules
+and default field values when you create CustomResourceDefinitions.
+
+To learn more, see the following resources:
+
+* [Validation rules](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules)
+* [Defaulting](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting)
+
+## Performance and latency {#performance-latency}
+
+This section describes recommendations for improving performance and reducing
+latency. In summary, these are as follows:
+
+* Consolidate webhooks and limit the number of API calls per webhook.
+* Use audit logs to check for webhooks that repeatedly do the same action.
+* Use load balancing for webhook availability.
+* Set a small timeout value for each webhook.
+
+### Improve latency in mutating webhooks {#improve-latency-mutating-webhooks}
+
+Mutating admission webhooks are called in sequence. Depending on the mutating
+webhook setup, some webhooks might be called multiple times. Every mutating
+webhook call adds latency to the admission process. This is unlike validating
+webhooks, which get called in parallel. 
+
+When designing your mutating webhooks, consider your latency requirements and
+tolerance. The more mutating webhooks there are in your cluster, the greater the
+chance of latency increases. 
+
+Consider the following to reduce latency:
```

Comment:
Should we advice how to measure this latency? Is there any articles about it already?

@jpbetz do you know?

Reply:
yeah. Also distributed tracing enabled on webhooks will be handy to see what's going on there